pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -486,11 +493,14 @@ func validateTasksInputFrom(tasks []PipelineTask) (errs *apis.FieldError) {
 	return errs
 }
 
-func validateWhenExpressions(tasks []PipelineTask) (errs *apis.FieldError) {
+func validateWhenExpressions(tasks []PipelineTask, finalTasks []PipelineTask) (errs *apis.FieldError) {
 	for i, t := range tasks {
 		errs = errs.Also(validateOneOfWhenExpressionsOrConditions(t).ViaFieldIndex("tasks", i))
```

Comment:
maybe this is overkill, but now that we're allowing when expressions in finally, does it make sense to add a test that ensures that conditions aren't allows in when expressions? (just ignore me if this already exists)

Reply:
> that conditions aren't allows in when expressions 

did you mean "that conditions aren't allowed in finally"? if so, that's already tested in:

https://github.com/tektoncd/pipeline/blob/075910e6a0e7e40fb01530aab05d4ec8ac5433a7/pkg/apis/pipeline/v1beta1/pipeline_validation_test.go#L2001-L2014